lightly/api/api_workflow_compute_worker.py

```diff
@@ -155,14 +165,82 @@ def get_scheduled_compute_worker_runs(
             dataset_id=self.dataset_id
         )
 
+    def get_scheduled_run_by_id(self, scheduled_run_id: str) -> DockerRunScheduledData:
+        """Returns the schedule run data given the id of the scheduled run.
+
+        TODO (MALTE, 09/2022): Have a proper API endpoint for doing this.
+        """
+        try:
+            run: DockerRunScheduledData = next(
+                run
+                for run in retry(
+                    lambda: self._compute_worker_api.get_docker_runs_scheduled_by_dataset_id(
+                        self.dataset_id
+                    )
+                )
+                if run.id == scheduled_run_id
+            )
+            return run
+        except StopIteration:
```

Comment:
> It also could be a 403/forbidden error

Not really, as it first gets all scheduled runs for you own dataset. This won't fail, but return an empty list in worst case. Thus this will raise a `ScheduledRunUnknown` for the case that the Scheduled run exists, just for another dataset.

Of course, it may happen that you get an "Access denied/403" because you have the wrong self.dataset_id, but this is another case.